tools/cmake/cores/Cortex-M4.cmake

```
@@ -12,11 +12,13 @@ elseif(${MBED_TOOLCHAIN} STREQUAL "ARM")
         "-mcpu=cortex-m4"
         "-mfpu=none"
     )
-#Necessary as the linker does not always detect
-#the architecture from the objectfiles correctly.
-    list(APPEND link_options
-        "--cpu=Cortex-M4.no_fp"
-    )
+    if(deprecated_system_processor)
+        # Necessary as the linker does not always detect
+        # the architecture from the object files correctly.
+        list(APPEND link_options
+            "--cpu=Cortex-M4.no_fp"
+        )
+    if(deprecated_system_processor)
```

Minor correction for this if block.
```suggestion
    endif()
```